wpilibc/src/main/native/include/frc/smartdashboard/ListenerExecutor.h

```diff
@@ -0,0 +1,41 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2019 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+#pragma once
+
+#include <functional>
+#include <vector>
+
+#include <wpi/mutex.h>
+
+namespace frc {
+/**
+ * An executor for running listener tasks posted by Sendable listeners
+ * synchronously from the main application thread.
+ *
+ * @see Sendable
+ */
+class ListenerExecutor {
```

Comment:
Since ListenerExecutor is package-private in Java, we could do something equivalent by putting the C++ ListenerExecutor in a `frc::detail` namespace. `namespace frc::detail {` is valid syntax, by the way.

Reply:
done.